Improve DDlog integration

ryzhyk opened the pull request:
These changes require the latest `northd` branch of ddlog:

    - add missing ovn.rs and ovn.dl files
    - move latest ovn_north.dl from https://github.com/ryzhyk/differential-datalog to the OVS repository;
      it will be maintained here from now on
    - link ovn_northd_cli against `libovenvswitch.a` and `libovn.a`
    - fix `ovn/northd/automake.mk` to compile `ovn_northd.dl` instead of
      `OVN_Northbound.dl` (the latter is a library used by the former)
    - Make in6_generate_lla() no longer inline
    - revert export of ipv6_multicast_to_ethernet (it is not currently needed by ddlog)
    - add a dependency to force ddlog compilation before ovn-northd.c (since ovn-northd.c includes a ddlog-generated header)
    - add ddlog test files (but still no proper test automation)

ryzhyk committed:
packets: Make in6_generate_lla() no longer inline.

This is needed by ddlog, but it can't handle inline functions.

ryzhyk committed:
fix ddlog build + update to the latest `ovn_northd.dl`

- add missing ovn.rs and ovn.dl files
- move latest ovn_north.dl from https://github.com/ryzhyk/differential-datalog to the OVS repository;
  it will be maintained here from now on
- link ovn_northd_cli against `libovenvswitch.a` and `libovn.a`
- fix `ovn/northd/automake.mk` to compile `ovn_northd.dl` instead of
  `OVN_Northbound.dl` (the latter is a library used by the former)

ryzhyk committed:
Revert "packets: Make ipv6_multicast_to_ethernet() no longer inline."

This reverts commit 7d8490b82d091ae5269fc751825cda377cfc394a.

ryzhyk committed:
include correct ddlog header in ovn-northd.c

OVN_Northbound_ddlog.h -> ovn_northd_ddlog.h

ryzhyk committed:
add a dependency to force ddlog compilation before ovn-northd.c

ryzhyk committed:
ovn: add dependency ovn/northd/ovn-northd.c: ovn/northd/ovn_northd_ddlog/ovn_northd_ddlog.h

is there a better way to force ddlog to run before compiling
ovn-northd.c

ryzhyk committed:
ovn: ddlog test files

temporary setup until ddlog tests are properly integrated in the OVS
test suite

ryzhyk committed:
ovn: pre-acl flows

justinpettit reviewed (changes requested):
Can you take a look at these changes and apply them if you agree?

Thanks,

--Justin

```
diff --git a/ovn/northd/automake.mk b/ovn/northd/automake.mk
index 1ef9bf37e3e1..7db3a8819731 100644
--- a/ovn/northd/automake.mk
+++ b/ovn/northd/automake.mk
@@ -49,8 +49,7 @@ ovn/northd/OVN_Southbound.dl: ovn/ovn-sb.ovsschema
                                -k Meter.name                   \
                                > $@
 
-CLEANFILES += ovn/northd/OVN_Northbound.dl ovn/northd/OVN_Southbound.dl \
-                         ovn/northd/OVN_Northbound.dl ovn/northd/OVN_Northbound.dl
+CLEANFILES += ovn/northd/OVN_Northbound.dl ovn/northd/OVN_Southbound.dl
 
 ovn/northd/ovn_northd_ddlog/target/debug/ovn_northd_cli: \
        ovn/northd/ovn_northd.dl         \
@@ -71,5 +70,6 @@ ovn_northd_ovn_northd_LDADD += \
 
 CLEANFILES += \
        ovn/northd/ovn_northd_ddlog/target/debug/libovn_northd_ddlog.la \
-       ovn/northd/ovn_northd_ddlog/ovn_northd_ddlog.h
+       ovn/northd/ovn_northd_ddlog/ovn_northd_ddlog.h \
+       ovn/northd/ovn_northd_ddlog/target/debug/ovn_northd_cli
 endif
```

justinpettit reviewed (commented):
I was able to edit the previous comment, so you can ignore this one.

justinpettit reviewed (changes requested):
Can you reorder commits and put this at the beginning?  I'll eventually just want to get rid of that commit entirely.

justinpettit reviewed (changes requested):
Can you merge this into the commit "fix ddlog build + update to the latest `ovn_northd.dl`", since that one just comments out the code.  I don't think breaking this across a couple of commits aids in anything.

justinpettit reviewed (changes requested):
Can you merge this into "fix ddlog build + update to the latest `ovn_northd.dl`"?

justinpettit reviewed (commented):
Doesn't this happen implicitly with the following?

```
ovn_northd_ovn_northd_SOURCES += \
    ovn/northd/ovn_northd_ddlog/ovn_northd_ddlog.h
```

ryzhyk committed:
fixes to ovn/northd/automate.mk

- The same file was listed in CLEANFILES three times
- added ovn_northd_cli to CLEANFILES

justinpettit commented:
I merged these changes into my branch with a few minor changes and squashes, and pushed it.
